ST_resgroups.c

```diff
@@ -185,124 +181,124 @@ void rgroup_PartResources(void)
         /* Get the maximum number of viable seeds from the seedbank that can
-         establish this year*/
-	IntU i;
-	RealF sum = 0.;
-	SpeciesType *s = Species[sp];
-
-	for (i = 1; i <= s->viable_yrs; i++)
-		sum += s->seedprod[i - 1] / pow(i, s->exp_decay);
-
-	return sum;
+     establish this year*/
+    IntU i;
+    RealF sum = 0.; //sum of the viable seedbank
+    SpeciesType *s = Species[sp];
+
+    for (i = s->viable_yrs; i > 0; i--)
+    {
+        //printf("Species name=%s , seedprod before decay s->seedprod[%hu]= %d \n", s->name, i, s->seedprod[i]);
+        s->seedprod[i] = s->seedprod[i] / pow(i, s->exp_decay);
+        //printf("Species name=%s , seedprod after decay s->seedprod[%hu]= %d \n", s->name, i, s->seedprod[i]);
+    }    
+       
+    for (i = 0; i <= s->viable_yrs; i++)
+    {
+        sum += s->seedprod[i];
+    }
+    //printf("sum =%f \n",sum);
+    
+    return  sum;
 }
 
-static void _add_annual_seedprod(SppIndex sp, RealF pr)
-{
-	/*======================================================*/
-	/* Add seeds to the seedbank this year and increment viable years for seeds.
-         * negative pr means add 0 seeds to seedbank */
-
-	SpeciesType *s = Species[sp];
-	IntU i;
-// incrementing the number of viable years
-	for (i = s->viable_yrs - 1; i > 0; i--)
-	{
-		s->seedprod[i] = s->seedprod[i - 1];
-	}
-
-	//compare pr (actual value or -1) to zero, if less than zero, then make it zero. If greater than 0
-	//than multiple max_seed_estab * exp(-pr)
-	s->seedprod[i] = LT(pr, 0.) ? 0. : s->max_seed_estab * exp(-pr);
-	//s->seedprod[i] = LT(pr, 0.) ? 0. : s->max_seed_estab * 1/pr ;
-	//s->seedprod[i] = LT(pr, 0.) ? 0. : s->max_seed_estab * s->relsize;
+static void _add_annual_seedprod(SppIndex sp, RealF lastyear_relsize) {
+    /*======================================================*/
+    /* Add seeds to the seedbank this year and increment viable years for seeds.*/
+
+    SpeciesType *s = Species[sp];
+    IntU i;
+
+    //incrementing the number of viable years
+    for (i = s->viable_yrs - 1; i > 0; i--) 
+    {
+        s->seedprod[i] = s->seedprod[i - 1];
+    }
+    
+   // printf("Species name=%s ,old Array array 0 index i=%u, value =%hu \n", s->name, i, s->seedprod[i]);
+
+    //If the current year is year 1 of the simulation, then the number of seeds added is a random number draw between
+    //1 and the maximum number of seedlings that can establish. Otherwise, this year's seed production is a function of the number 
+    //of seeds produced per unit biomass, max biomass, and last year's species relative size.
+    if (Globals.currYear == 1) {
+        s->seedprod[i] = RandUniRange(1, s->pseed);
+       //printf("Species name=%s ,currYear =1 so new calculated value s->seedprod[%u]= %hu , s->max_seed_estab =%hu\n", s->name, i, s->seedprod[i], s->max_seed_estab);
+    
+    } else {
+    	s->seedprod[i] = (IntU)(s->pseed * s->mature_biomass * lastyear_relsize);
```

Comment:
The value of `i` here and in line 295 is 0. Maybe write `s->seedprod[0]` instead of `s->seedprod[i]` to make the code easier to read?

Reply:
Good idea, fixed with commit https://github.com/DrylandEcology/STEPWAT2/commit/5cc6bad925b78f728a30fc3019609aad256606fa